pkg/ccl/crosscluster/logical/lww_row_processor_test.go

```diff
@@ -301,3 +306,165 @@ func BenchmarkLWWInsertBatch(b *testing.B) {
+		require.NoError(t, err)
+
+		if useKVProc {
+			rp, err = newKVRowProcessor(ctx,
+				&execinfra.ServerConfig{
+					DB:           s.InternalDB().(descs.DB),
+					LeaseManager: s.LeaseManager(),
+				}, &eval.Context{
+					Codec:    s.Codec(),
+					Settings: s.ClusterSettings(),
+				}, map[descpb.ID]sqlProcessorTableConfig{
+					dstDesc.GetID(): {
+						srcDesc: srcDesc,
+					},
+				},
+				rp.(*sqlRowProcessor))
+			require.NoError(t, err)
+		}
+		return tableNameDst, rp, func(timestamp hlc.Timestamp, datums ...interface{}) roachpb.KeyValue {
+			kv := replicationtestutils.EncodeKV(t, s.Codec(), srcDesc, datums...)
+			kv.Value.Timestamp = timestamp
+			return kv
+		}
+	}
+
+	insertRow := func(rp RowProcessor, keyValue roachpb.KeyValue, prevValue roachpb.Value) error {
+		return s.InternalDB().(isql.DB).Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
+			_, err := rp.ProcessRow(ctx, txn, keyValue, prevValue)
+			return err
+		})
+	}
+
+	timeNow := hlc.Timestamp{WallTime: timeutil.Now().UnixNano()}
+	timeOneDayForward := hlc.Timestamp{WallTime: timeutil.Now().Add(time.Hour * 24).UnixNano()}
+	row1 := []interface{}{1, "row1"}
+	row2 := []interface{}{1, "row2"}
+	row3 := []interface{}{1, "row3"}
+
+	testutils.RunTrueAndFalse(t, "useKVProc", func(t *testing.T, useKVProc bool) {
+		testutils.RunTrueAndFalse(t, "optimistic_insert", func(t *testing.T, optimisticInsert bool) {
+
+			t.Run("cross-cluster-write", func(t *testing.T) {
+				runner.Exec(t, fmt.Sprintf("SET CLUSTER SETTING logical_replication.consumer.try_optimistic_insert.enabled=%t", optimisticInsert))
+				defer runner.Exec(t, "RESET CLUSTER SETTING logical_replication.consumer.try_optimistic_insert.enabled")
+				tableNameDst, rp, encoder := setup(t, useKVProc)
+
+				runner.Exec(t, fmt.Sprintf("INSERT INTO %s VALUES ($1, $2)", tableNameDst), row1...)
+
+				keyValue2 := encoder(timeOneDayForward, row2...)
+				require.NoError(t, insertRow(rp, keyValue2, roachpb.Value{}))
+
+				expectedRows := [][]string{
+					{"1", "row2"},
+				}
+				runner.CheckQueryResults(t, fmt.Sprintf("SELECT * from %s", tableNameDst), expectedRows)
+			})
+
+			t.Run("sql-conflict-incoming-newer", func(t *testing.T) {
+				runner.Exec(t, fmt.Sprintf("SET CLUSTER SETTING logical_replication.consumer.try_optimistic_insert.enabled=%t", optimisticInsert))
+				defer runner.Exec(t, "RESET CLUSTER SETTING logical_replication.consumer.try_optimistic_insert.enabled")
+				tableNameDst, rp, encoder := setup(t, useKVProc)
+
+				keyValue1 := encoder(timeNow, row1...)
+				require.NoError(t, insertRow(rp, keyValue1, roachpb.Value{}))
+
+				keyValue2 := encoder(timeOneDayForward, row2...)
+				require.NoError(t, insertRow(rp, keyValue2, roachpb.Value{}))
+
+				expectedRows := [][]string{
+					{"1", "row2"},
+				}
+				runner.CheckQueryResults(t, fmt.Sprintf("SELECT * from %s", tableNameDst), expectedRows)
+			})
+
+			t.Run("sql-conflict-incoming-older", func(t *testing.T) {
```

Comment:
the point of this test case is to check behavior after a rangefeed re-transmission, and i've realized you can just check this in the case above and delete this case. 

After you check 
```
expectedRows := [][]string{
					{"1", "row2"},
```

above, just re-insert `kv1` and reassert the same expected rows.

Reply:
Done